pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -95,6 +124,13 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	if !provisioning && !completed {
 		userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
 		if err != nil {
+			err := r.setStatusFailedToReadUserApprovalPolicy(instance, "Failed to read ConfigMap value " + config.ToolchainConfigMapUserApprovalPolicy)
```

Comment:
Here and in all other places where you set the failed status. Let's pass the error string as the message. If you want you can wrap it in your own message but don't hide it.
For example:
https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L133
https://github.com/codeready-toolchain/member-operator/blob/master//pkg/controller/useraccount/useraccount_controller.go#L217:32
So if something failed then we will get the original error message in the status.